src/main/java/com/zombiecastlerush/role/Player.java

```diff
@@ -23,13 +24,27 @@ public Player(String name, Room room) {
      */
     public boolean moveTo(String roomName){
         Room targetRoom = this.canMoveToRoom(roomName);
-        if(targetRoom != null){
+        Challenge currRoomChallenge = super.getCurrentPosition().getChallenge();
```

Comment:
What's the difference between calling `super.getCurrentPosition()` and `this.getCurrentPosition()`? If there's no difference, I think it would be better to use `this`, as it is clearer to reader what you intend.

Reply:
will change to this